server/catgenome/src/main/java/com/epam/catgenome/controller/dataitem/DataItemController.java

```diff
@@ -123,4 +129,17 @@
             throws IOException {
         return Result.success(dataItemSecurityService.findFileByBioItemId(id));
     }
+
+    @GetMapping("/dataitem/{id}/download")
+    @ApiOperation(
+            value = "Downloads a file specified by biological item id",
+            notes = "Downloads a file specified by biological item id",
+            produces = MediaType.APPLICATION_OCTET_STREAM_VALUE)
+    public void downloadFileByBiologicalItemId(@PathVariable(value = "id") final Long id,
+                                               final HttpServletResponse response) throws IOException {
+        final BiologicalDataItem biologicalDataItem = dataItemSecurityService.findFileByBioItemId(id);
+        final InputStream content = dataItemSecurityService.loadFileContent(biologicalDataItem);
```

Comment:
how about to get from `dataItemSecurityService` not only stream but object that will  represent a stream and also a name of the file, I suppose it will be mo convenient to move this logic to the manager level, because there could be some specific logic (f.e. for different file location S3, Azure)

Reply:
done